### ISSUE TITLE
Sytemd Proxy not apply

### ISSUE THREAD

**Issue by jynolen** [saltstack-formulas/docker-formula#230]
tldr;
If proxy is provide to pillar the formula should create a file in 
`/etc/systemd/system/docker.service.d`
with content
```
[Service]
Environment="HTTP_PROXY=http://proxy.example.com:80/"
```
=====
Hello,
It seems that if the system is using systemd and if we provide proxy var to the pillar the proxy conf is not applied as docker configuration said :
[https://docs.docker.com/config/daemon/systemd/](https://docs.docker.com/config/daemon/systemd/)
**You cannot configure these environment variables using the daemon.json file.**

**If you are behind an HTTP or HTTPS proxy server, for example in corporate settings, you need to add this configuration in the Docker systemd service file.**

For now, the workarround is to have another state to file.managed the systemd file. but it could be great to insert it to the formula.
Should produce

**Comment by noelmcloughlin** [comment]
Fixed in merged #256
